### DISCUSSION TITLE
How to catch an event in orchard

### DISCUSSION THREAD

**Discussion by AmadeuAntunes** [OrchardCMS/OrchardCore#8463] [Q&A]
Hello, is there any way for I catch the  CloningAsync(CloneContentContext context) event when some content is about to get cloned.

When someone click on Clone content item get an event notification for can add some actions there?

![image](https://user-images.githubusercontent.com/47931101/106300960-058d0b00-624f-11eb-9b0b-3d64a69c18ff.png)

**Comment by tropcicstefan** [comment]
You should look into [ContentPartHandlers](https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs). 

Or into feature of Workflows, but I think there is no ClonedEvent.

**Reply by AmadeuAntunes** [reply]
I found this event in IContentHandler

**Reply by hishamco** [reply]
The handler contains many methods that you could subscripe to

**Reply by AmadeuAntunes** [reply]
@hishamco Can you show me a simple exemple

**Reply by hishamco** [reply]
You need to create a class that inherits from `ContentHandlerBase` then override the event that it needs to execute your code when it's tiggered

Hint: don't forgot to register your class into the DI as the following:

```csharp
 services.AddScoped<IContentHandler, CloneContentHandler>();
```
